File: src/main/java/org/jukeboxmc/form/element/StepSlider.java
package org.jukeboxmc.form.element;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.jukeboxmc.form.CustomForm;

import java.util.ArrayList;
import java.util.List;

/**
 * @author GoMint
 * @version 1.0
 */
public class StepSlider extends Element{

    private final CustomForm form;
    private final List<String> steps = new ArrayList<>();
    private int defaultStep = 0;

    public StepSlider( CustomForm form, String id, String text ) {
        super( id, text );
        this.form = form;
    }

    public StepSlider addStep( String step ) {
        this.steps.add( step );
        this.form.setDirty();
        return this;
    }

    public StepSlider addStep( String step, boolean defaultStep ) {
        if ( defaultStep ) {
            this.defaultStep = this.steps.size();
        }

        this.steps.add( step );
        this.form.setDirty();
        return null;
    }

    @Override
    public JSONObject toJSON() {
        JSONObject obj = super.toJSON();
        obj.put( "type", "step_slider" );

        JSONArray jsonSteps = new JSONArray();
        jsonSteps.addAll( this.steps );

        obj.put( "steps", jsonSteps );
        obj.put( "default", this.defaultStep );
        return obj;
    }

    @Override
    public Object getAnswer( Object answerOption ) {
        String answer = (String) answerOption;
        this.defaultStep = this.steps.indexOf( answer );
        return answer;
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/element/StepSlider.kt
package org.jukeboxmc.api.form.element

import com.google.gson.JsonArray
import com.google.gson.JsonObject
import org.jukeboxmc.api.form.CustomForm

/**
 * @author Kaooot
 * @version 1.0
 */
class StepSlider(private val form: CustomForm, id: String, text: String) : Element(id, text) {

    private val steps = arrayListOf<String>()
    private var defaultStep = 0

    override fun toJSON(): JsonObject {
        val jsonObject = super.toJSON()
        jsonObject.addProperty("type", "step_slider")
        val steps = JsonArray()
        for (step in this.steps) {
            steps.add(step)
        }
        jsonObject.add("steps", steps)
        jsonObject.addProperty("default", this.defaultStep)
        return jsonObject
    }

    override fun getAnswer(answerOption: Any): Any {
        val answer = answerOption as String
        this.defaultStep = this.steps.indexOf(answer)
        return answer
    }

    fun addStep(step: String): StepSlider {
        this.steps.add(step)
        this.form.setDirty()
        return this
    }

    fun addStep(step: String, defaultStep: Boolean): StepSlider {
        if (defaultStep) {
            this.defaultStep = this.steps.size
        }
        this.steps.add(step)
        this.form.setDirty()
        return this
    }
}